modules/core/src/effects/mask/utils.js

```diff
@@ -62,6 +62,13 @@ export function getMaskBounds({layers, viewport}) {
  * Compute viewport to render the mask into, covering the given bounds
  */
 export function getMaskViewport({bounds, viewport, width, height}) {
+  if (bounds[2] <= bounds[0] || bounds[3] <= bounds[1]) {
+    return null;
+  }
```

Comment:
I would add a comment here explaining why this padding is necessary.
```suggestion
  }
  
  // Single pixel border to prevent mask bleeding at edge of texture
```